tests/NLog.UnitTests/LayoutRenderers/CallSiteTests.cs

```diff
@@ -1040,6 +1040,72 @@ public void CallSiteShouldWorkEvenInlined()
             Assert.Equal("NLog.UnitTests.LayoutRenderers.CallSiteTests.CallSiteShouldWorkEvenInlined", callSite);
         }
 
+#if ASYNC_SUPPORTED
+        [Fact]
+        public void LogAfterAwait_CleanNamesOfAsyncContinuationsIsTrue_ShouldCleanMethodName()
+        {
+            LogManager.Configuration = CreateConfigurationFromString(@"
+                <nlog>
+                    <targets><target name='debug' type='Debug' layout='${callsite:classname=false:cleannamesofasynccontinuations=true}' /></targets>
+                    <rules>
+                        <logger name='*' levels='Debug' writeTo='debug' />
+                    </rules>
+                </nlog>");
+
+            AsyncMethod5().GetAwaiter().GetResult();
+
+            AssertDebugLastMessage("debug", nameof(AsyncMethod5));
+        }
+
+        [Fact]
+        public void LogAfterAwait_CleanNamesOfAsyncContinuationsIsTrue_ShouldCleanClassName()
+        {
+            LogManager.Configuration = CreateConfigurationFromString(@"
+                <nlog>
+                    <targets><target name='debug' type='Debug' layout='${callsite:classname=true:includenamespace=true:cleannamesofasynccontinuations=true}' /></targets>
+                    <rules>
+                        <logger name='*' levels='Debug' writeTo='debug' />
+                    </rules>
+                </nlog>");
+
+            AsyncMethod5().GetAwaiter().GetResult();
+
+            AssertDebugLastMessage("debug", typeof(CallSiteTests).FullName + "." + nameof(AsyncMethod5));
+        }
+
+        [Fact]
+        public void LogAfterAwait_CleanNamesOfAsyncContinuationsIsFalse_ShouldNotCleanNames()
+        {
+            LogManager.Configuration = CreateConfigurationFromString(@"
+                <nlog>
+                    <targets><target name='debug' type='Debug' layout='${callsite:includenamespace=true:cleannamesofasynccontinuations=false}' /></targets>
+                    <rules>
+                        <logger name='*' levels='Debug' writeTo='debug' />
+                    </rules>
+                </nlog>");
+
+            AsyncMethod5().GetAwaiter().GetResult();
+
+            var loggedCallsite = GetDebugLastMessage("debug");
+            Assert.Contains("<", loggedCallsite);
+            Assert.Contains(">", loggedCallsite);
+            Assert.Contains("+", loggedCallsite);
+            Assert.Contains("MoveNext", loggedCallsite);
```

Comment:
If it's possible, please test with full string and equals.

Reply:
OK, I have done that now. I'm afraid it can be somewhat brittle, though, since the exact compiler-generated method name varies somewhat, ie. probably when other members are added to the method. Must be prapared to update this test when that happens.